src/abTestingInfra/emitter.ts

```diff
@@ -0,0 +1,72 @@
+import EventEmitter from 'eventemitter3';
+
+const emitter = new EventEmitter();
+
+export const EXPERIMENT_PLAYED = 'EXPERIMENT_PLAYING';
+export const EXPERIMENT_WON = 'EXPERIMENT_WON';
+
+declare global {
+  interface Window {
+    experiments: Map<string, boolean>;
+  }
+}
+
+if (typeof window !== 'undefined') {
+  window.experiments = new Map<string, boolean>();
+}
+
+export const recordPlay = (
+  experimentName: string,
+  variantName: string,
+): void => {
+  if (typeof window === 'undefined') {
+    return;
+  }
+
+  if (!experimentName) {
+    throw new Error('Experiment name is undefined');
+  }
+
+  if (!variantName) {
+    throw new Error('Variant name is undefined');
+  }
```

Comment:
We can probably get away with using the `assert` built-in lib for this, i.e. `assert.ok(variantName, 'Variant name is undefined')`

Reply:
Done